moped-database/migrations/1691605248601_council-district-trigger/up.sql

```diff
@@ -0,0 +1,157 @@
+-- add spatial index to this layer
+CREATE INDEX layer_council_district_geom_idx
+  ON layer_council_district
+  USING GIST (geography);
+
+-- create table to hold feature-district associations
+CREATE TABLE public.features_council_districts (
+    id serial PRIMARY KEY,
+    feature_id integer NOT NULL,
+    council_district_id integer NOT NULL,
+    FOREIGN KEY ("council_district_id") REFERENCES "public"."layer_council_district" ("council_district")
+        ON UPDATE CASCADE ON DELETE CASCADE
+);
+
+ALTER TABLE public.features_council_districts
+    ADD CONSTRAINT unique_feature_council_district UNIQUE ("feature_id", "council_district_id");
+
+CREATE INDEX features_council_district_feature_id_idx ON features_council_districts (feature_id);
+CREATE INDEX features_council_district_council_district_id_idx ON features_council_districts (council_district_id);
+
+-- create functions to mange point and line type associations
+CREATE OR REPLACE FUNCTION public.update_line_council_district()
+    RETURNS TRIGGER
+    LANGUAGE plpgsql
+    AS $$
+BEGIN
+    -- currently the moped editor never updates a geometry, it marks a feature as is_deleted
+    -- and inserts a new feature. so we should only ever see a geometry update from some
+    -- ad-hoc developer intervention
+    IF TG_OP = 'UPDATE' AND ST_Equals(NEW.geography::geometry, OLD.geography::geometry) THEN
+        -- nothing to do
+        RETURN NEW;
+    END IF;
+    -- delete previous district associations
+    -- again, because the moped editor only inserts new geometries, there will typically
+    -- not be any old feature-district associations to delete
+    DELETE FROM features_council_districts WHERE feature_id = NEW.id;
+    -- insert new district associations
+    WITH inserts_todo AS (
+        SELECT
+            NEW.id AS feature_id,
+            districts.council_district AS council_district_id
+        FROM
+            layer_council_district AS districts
+        WHERE
+            ST_Intersects(districts.geography, NEW.geography)
+        OR
+            ST_Crosses(districts.geography::geometry, NEW.geography::geometry)
+    )
+        INSERT INTO features_council_districts (feature_id, council_district_id)
+            SELECT * FROM inserts_todo;
+    RETURN NEW;
+    END;
+$$;
+
+CREATE OR REPLACE FUNCTION public.update_point_council_district()
```

Comment:
i opted for separate point vs line functions, although it would actually be possible to combine these using an `IF` block that checks the special `TG_TABLE_NAME` variable....

Reply:
We've been thinking about this in VZ a lot lately too. I think separate functions is a good call since we don't have a good feel for if this logic will be recycled yet (or at least I don't 😅)